### DISCUSSION TITLE
Android apk

### DISCUSSION THREAD

**Discussion by icestorms** [esnet/iperf#1537] [General]
Can you good guys release apks  of recent iperf3 stable version such as 3.9 or 3.13 for non-coder like me.

**Comment by icestorms** [comment]
Can you also update Magic iperf apk to a new iperf version and enable it running in the background ?

**Comment by davidBar-On** [comment]
As far as I know, currently there is no one that builds up to date iperf3 versions for Android.  [This site](https://github.com/KnightWhoSayNi/android-iperf) maintained iperf3 for Android up to version 3.10.1.

For Magic Iperf you should contact with the APK developer.

**Comment by tyerllwendland512** [comment]
As a user, it would be great if the good guys could release APKs of recent stable versions of iPerf3 like 3.9 or 3.13. Having access to updated versions would be helpful, especially for non-coders like myself. It would make it easier to install and use the application without the need for technical knowledge. I appreciate any support in making the latest [APK](https://trafficriderapp.com/) iPerf3 versions available through APK releases.

**Comment by bmah888** [comment]
Just to be clear, ESnet (maintainers of iperf3) only release source code, through source code tarballs and the GitHub repo. It's up to operating system packagers and/or third parties to build and distribute iperf3 binaries for a variety of different platforms.

**Comment by davidBar-On** [comment]
I have created a new repository with 3.14 binaries: [https://github.com/davidBar-On/android-iperf3](https://github.com/davidBar-On/android-iperf3). (The repository is based on [KnightWhoSayNi](https://github.com/KnightWhoSayNi/android-iperf/) repository that built iperf3 for android up to version 3.10.1).

My testing capabilities are limited, so it will be  **a great help to test the binaries** and make sure the build process is o.k.  Any other comment is welcome.

I plan to add new iperf3 versions when they will be available.

**Comment by davidBar-On** [comment]
I have added **Multi-thread** beta3 (iperf3.13-mt-beta3) Android build and binaries to  [https://github.com/davidBar-On/android-iperf3](https://github.com/davidBar-On/android-iperf3).   Building the multi-thread version includes source patches, as Android NDK does not support `pthread_cancel()`.

**Comment by Abhi5h3k** [comment]
I’ve been working on something similar — an Android frontend for iPerf3 called [CellularLab](https://github.com/Abhi5h3k/CellularLab) that wraps native iPerf via JNI.

It supports TCP, UDP, bidirectional tests, command mode, and even AI-powered log analysis using Gemini.

I also wrote about how I “vibe-coded” it with AI tools:
📄 [How I Accidentally Vibe-Coded an Android iPerf3 App with AI](https://medium.com/@abhishekbhardwajca/how-i-accidentally-vibe-coded-an-android-iperf3-app-with-ai-b8eea31b7773)

Might be useful for anyone looking for mobile test automation or ideas on integrating iPerf in non-traditional ways. Happy to collab or get feedback too. 🚀

![CellularLab](https://github.com/user-attachments/assets/e34651f3-0781-4183-a53c-0bf229d036a2)
